### PR TITLE
Fix for the Linux Kernel v6.2 (Fedora 37/38)

### LINKED ISSUE #247: Support kernel  6.2.9 as in the upcoming Fedora 38
Now Fedora 37 already has kernel `6.2.9-200.fc37.x86_64` as it will be in the upcoming release of Fedora 38.

The build is failing like this:
```
make -C src
make[1]: Entering directory '/home/elastio/elastio-snap/src'
if [ ! -f kernel-config.h ] || tail -1 kernel-config.h | grep -qv '#endif'; then mkdir configure-tests/feature-tests/build; ./genconfig.sh "6.2.9-200.fc37.x86_64" "-w"; fi;
generating configurations for kernel-6.2.9-200.fc37.x86_64
make[2]: Entering directory '/home/elastio/elastio-snap/src/configure-tests/feature-tests'
make[3]: Entering directory '/usr/src/kernels/6.2.9-200.fc37.x86_64'
make[3]: Leaving directory '/usr/src/kernels/6.2.9-200.fc37.x86_64'
make[2]: Leaving directory '/home/elastio/elastio-snap/src/configure-tests/feature-tests'
xargs: warning: options --max-args and --replace/-I/-i are mutually exclusive, ignoring previous --max-args value
performing configure test: HAVE_BDEVNAME - not present
performing configure test: HAVE_ALLOC_DISK - not present
performing configure test: HAVE_BDEV_STACK_LIMITS - not present
performing configure test: HAVE_BDEV_IS_PARTITION - present
performing configure test: HAVE_BDEV_NR_SECTORS - present
performing configure test: HAVE_ADD_DISK_INT - present
performing configure test: HAVE_BDOPS_OPEN_INODE - not present
performing configure test: HAVE_BDOPS_OPEN_INT - not present
performing configure test: HAVE_BDOPS_SUBMIT_BIO_UINT - not present
performing configure test: HAVE_BDOPS_SUBMIT_BIO - present
performing configure test: HAVE_BD_SUPER - present
performing configure test: HAVE_BIO_ALLOC_BIOSET_5 - present
performing configure test: HAVE_BIO_BI_BDEV_BD_DISK - present
performing configure test: HAVE_BIO_BI_REMAINING - not present
performing configure test: HAVE_BIO_BI_BDEV - present
performing configure test: HAVE_BIO_COPY_DEV - not present
performing configure test: HAVE_BIO_BI_POOL - present
performing configure test: HAVE_BIO_ENDIO_INT - not present
performing configure test: HAVE_BIO_ENDIO_1 - present
performing configure test: HAVE_BIOSET_CREATE_3 - not present
performing configure test: HAVE_BIO_FREE_PAGES - present
performing configure test: HAVE_BIO_LIST - present
performing configure test: HAVE_BIO_REMAPPED - present
performing configure test: HAVE_BIO_SET_DEV - present
performing configure test: HAVE_BIOSET_INIT - present
performing configure test: HAVE_BIO_THROTTLED - not present
performing configure test: HAVE_BIO_SPLIT_4 - present
performing configure test: HAVE_BIOSET_NEED_BVECS_FLAG - present
performing configure test: HAVE_BLK_ALLOC_QUEUE_GFP_T - not present
performing configure test: HAVE_BLK_ALLOC_QUEUE_MK_REQ_FN_NODE_ID - not present
performing configure test: HAVE_BLKDEV_PUT_1 - not present
performing configure test: HAVE_BLK_MQ_MAKE_REQUEST - not present
performing configure test: HAVE_BLK_MQ_SUBMIT_BIO - not present
performing configure test: HAVE_BLKDEV_GET_BY_PATH - present
performing configure test: HAVE_BLK_SET_STACKING_LIMITS - present
performing configure test: HAVE_BLK_STATUS_T - present
performing configure test: HAVE_BVEC_MERGE_DATA - not present
performing configure test: HAVE_BVEC_ITER_ALL - present
performing configure test: HAVE_BVEC_ITER - present
performing configure test: HAVE_COMPOUND_HEAD - present
performing configure test: HAVE___DENTRY_PATH - not present
performing configure test: HAVE_DENTRY_PATH_RAW - present
performing configure test: HAVE_DISK_LIVE - present
performing configure test: HAVE_ENUM_REQ_OPF - not present
performing configure test: HAVE_D_UNLINKED - present
performing configure test: HAVE_ENUM_REQ_OP - present
performing configure test: HAVE_FILE_INODE - present
performing configure test: HAVE_GENERIC_MAKE_REQUEST - not present
performing configure test: HAVE_FMODE_T - present
performing configure test: HAVE_FOPS_FALLOCATE - present
performing configure test: HAVE_FREEZE_SUPER - present
performing configure test: HAVE_GENHD_FL_NO_PART_SCAN - not present
performing configure test: HAVE_GENHD_FL_NO_PART - present
performing configure test: HAVE_IOPS_FALLOCATE - not present
performing configure test: HAVE_INODE_LOCK - present
performing configure test: HAVE_KERNEL_READ_PPOS - present
performing configure test: HAVE_LOCKS_VERIFY_TRUNCATE - not present
performing configure test: HAVE_KERNEL_WRITE_PPOS - present
performing configure test: HAVE_KERN_PATH - present
performing configure test: HAVE_MAKE_REQUEST_FN_IN_QUEUE - not present
performing configure test: HAVE_MAKE_REQUEST_FN_INT - not present
performing configure test: HAVE_MAKE_REQUEST_FN_VOID - not present
performing configure test: HAVE_MERGE_BVEC_FN - not present
performing configure test: HAVE_MNT_WANT_WRITE - present
performing configure test: HAVE_NOTIFY_CHANGE_2 - not present
performing configure test: HAVE_NOOP_LLSEEK - present
performing configure test: HAVE_NOTIFY_CHANGE_3 - not present
performing configure test: HAVE_PART_NR_SECTS_READ - not present
performing configure test: HAVE_PATH_PUT - present
performing configure test: HAVE_PROC_CREATE_FN_FILE_OPERATIONS - not present
performing configure test: HAVE_PROC_CREATE_FN_PROC_OPS - present
performing configure test: HAVE_SB_START_WRITE - present
performing configure test: HAVE_SI_MEM_AVAILABLE - present
performing configure test: HAVE_STRUCT_PATH - present
performing configure test: HAVE_SUBMIT_BIO_WAIT - not present
performing configure test: HAVE_SUBMIT_BIO_1 - present
performing configure test: HAVE_SYS_OLDUMOUNT - not present
performing configure test: HAVE_TASK_STRUCT_TASK_WORKS_HLIST - not present
performing configure test: HAVE_THAW_BDEV_INT - not present
performing configure test: HAVE_TASK_STRUCT_STATE - present
performing configure test: HAVE_TASK_STRUCT_TASK_WORKS_CB_HEAD - present
performing configure test: HAVE_UAPI_MOUNT_H - present
performing configure test: HAVE_USER_PATH_AT - present
performing configure test: HAVE_VFS_UNLINK_2 - not present
performing configure test: HAVE_VFS_UNLINK_3 - not present
performing configure test: HAVE_UUID_H - present
performing configure test: HAVE_VFS_FALLOCATE - present
performing configure test: HAVE_VZALLOC - present
performing configure test: HAVE_WRITE_ZEROES - present
make[2]: Entering directory '/home/elastio/elastio-snap/src/configure-tests/feature-tests'
make[3]: Entering directory '/usr/src/kernels/6.2.9-200.fc37.x86_64'
make[3]: Leaving directory '/usr/src/kernels/6.2.9-200.fc37.x86_64'
make[2]: Leaving directory '/home/elastio/elastio-snap/src/configure-tests/feature-tests'
performing sys_mount lookup
performing sys_umount lookup
performing __x64_sys_mount lookup
performing __x64_sys_umount lookup
performing __arm64_sys_mount lookup
performing __arm64_sys_umount lookup
performing __change_memory_common lookup
performing sys_oldumount lookup
performing sys_call_table lookup
performing kfree lookup
performing blk_mq_submit_bio lookup
performing blk_alloc_queue lookup
performing get_super lookup
checking CONFIG_ARM64_PAGE_SHIFT
make -C /lib/modules/6.2.9-200.fc37.x86_64/build M=/home/elastio/elastio-snap/src modules
make[2]: Entering directory '/usr/src/kernels/6.2.9-200.fc37.x86_64'
  CC [M]  /home/elastio/elastio-snap/src/elastio-snap.o
/home/elastio/elastio-snap/src/elastio-snap.c: In function ‘elastio_snap_set_bio_ops’:
/home/elastio/elastio-snap/src/elastio-snap.c:300:9: error: implicit declaration of function ‘bio_set_op_attrs’ [-Werror=implicit-function-declaration]
  300 |         bio_set_op_attrs(bio, op, op_flags);
      |         ^~~~~~~~~~~~~~~~
cc1: some warnings being treated as errors
make[3]: *** [scripts/Makefile.build:252: /home/elastio/elastio-snap/src/elastio-snap.o] Error 1
make[2]: *** [Makefile:2025: /home/elastio/elastio-snap/src] Error 2
make[2]: Leaving directory '/usr/src/kernels/6.2.9-200.fc37.x86_64'
make[1]: *** [Makefile:14: default] Error 2
make[1]: Leaving directory '/home/elastio/elastio-snap/src'
make: *** [Makefile:32: driver] Error 2
Error: Process completed with exit code 2.
```

### PR DESCRIPTION
**Pull request by skypodolsky**
* Enable Fedora 38
* Added compat for `bio_set_op_attrs`
* Fixed kernel panic on dereferencing the wrong address

Closes #247

### CODE REVIEW

File: src/configure-tests/feature-tests/bio_set_op_attrs.c [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/*
+ * Copyright (C) 2022 Elastio Software Inc.
```
Comment (anelson) [inline src/configure-tests/feature-tests/bio_set_op_attrs.c:4]: Shouldn't this be 2023 now?

**Review by anelson — APPROVED** [review]

File: src/configure-tests/feature-tests/bio_set_op_attrs.c (hunk 1 above)
Comment (skypodolsky) [reply src/configure-tests/feature-tests/bio_set_op_attrs.c:4]: Good point! Silly copy-paste

Comment (skypodolsky) [reply src/configure-tests/feature-tests/bio_set_op_attrs.c]: Fixed

**Review by vsazhenyuk-softheme — APPROVED** [review]

**Comment by e-kov** [comment]
Please add `amazon2023` to the [Vagrantfile](https://github.com/elastio/elastio-snap/blob/d3afec9e5dc70b551207d7432272bcd55a37b9d2/.github/buildbox/Vagrantfile#L37-L49).

**Pull request closed without merging**